### PR TITLE
[PropertyInfo] Dont try to set null when argument is not nullable

### PR DESCRIPTION
**Pull request by Nyholm**
| Q             | A
| ------------- | ---
| Branch?       | 5.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Part of #38800
| License       | MIT
| Doc PR        | 

This will break up #38800 into two smaller pieces. 

If there is a method `setPreviousException(Throwable $t)` and we try to unserialise a value of ['previousException'=>null], we should not try to invoke this setter. It will cause an error.

### CODE REVIEW

**Comment by Nyholm** [comment]
The test `testThrowTypeErrorWithNullArgument` is failing for this. Why would the desired behavior be that we get an Error like `Expected argument of type "Throwable", "null" given`?

**Comment by jderusse** [comment]
If it's a bug fix, shouldn't it target 4.4?

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php [hunk 1]
```diff
@@ -388,6 +388,12 @@ public function getWriteInfo(string $class, string $property, array $context = [
             }
 
             $method = $reflClass->getMethod($methodName);
+            /** @var \ReflectionParameter $parameter */
+            $parameter = $method->getParameters()[0];
```
Comment (jderusse) [inline src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:392]: Are we sure that the method as at least one parameter?

**Review by jderusse — COMMENTED** [review]
What's about constructor, non nullable properties (typehinted in php 7.4) and mutators?.

**Comment by Nyholm** [comment]
> If it's a bug fix, shouldn't it target 4.4?

Last time I checked, the code was very different in 4.4 and 5.1. I can check again later.

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 2]
```diff
@@ -610,6 +610,7 @@ private function getWriteInfo(string $class, string $property, $value): Property
             'enable_magic_methods_extraction' => $this->magicMethodsFlags,
             'enable_constructor_extraction' => false,
             'enable_adder_remover_extraction' => $useAdderAndRemover,
+            'value' => $value,
```
Comment (xabbuh) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php:615]: Is this change needed?

Comment (Nyholm) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php:615]: Yes, The PropertyInfo needs to know if the value is null or not.

Comment (xabbuh) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php:615]: Can we add a test covering this case then?

**Comment by fabpot** [comment]
@Nyholm Can you finish this one? I think it should go in 5.2.

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php (hunk 2 above)
Comment (Nyholm) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php:615]: Tests are added!

File: src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php (hunk 1 above)
Comment (Nyholm) [reply src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php:392]: Yes, just a few lines above we check that the method has at least one argument. I added a test to show that too.

File: src/Symfony/Component/PropertyAccess/PropertyAccessor.php [hunk 3]
```diff
@@ -86,7 +86,7 @@ class PropertyAccessor implements PropertyAccessorInterface
      *                          to specify the allowed magic methods (__get, __set, __call)
      *                          or self::DISALLOW_MAGIC_METHODS for none
      */
-    public function __construct(/*int */$magicMethods = self::MAGIC_GET | self::MAGIC_SET, bool $throwExceptionOnInvalidIndex = false, CacheItemPoolInterface $cacheItemPool = null, bool $throwExceptionOnInvalidPropertyPath = true, PropertyReadInfoExtractorInterface $readInfoExtractor = null, PropertyWriteInfoExtractorInterface $writeInfoExtractor = null)
+    public function __construct(/*int */ $magicMethods = self::MAGIC_GET | self::MAGIC_SET, bool $throwExceptionOnInvalidIndex = false, CacheItemPoolInterface $cacheItemPool = null, bool $throwExceptionOnInvalidPropertyPath = true, PropertyReadInfoExtractorInterface $readInfoExtractor = null, PropertyWriteInfoExtractorInterface $writeInfoExtractor = null)
```
Comment (Nyholm) [inline src/Symfony/Component/PropertyAccess/PropertyAccessor.php:89]: This was suggested by fabbot.

Comment (nicolas-grekas) [reply src/Symfony/Component/PropertyAccess/PropertyAccessor.php:89]: let's revert it :)

**Review by nicolas-grekas — COMMENTED** [review]
Can you please rebase to target 5.3?

**Comment by fabpot** [comment]
@Nyholm What's the status here?

**Comment by fabpot** [comment]
@Nyholm Sorry to ping you again on this one :)

**Comment by Nyholm** [comment]
I thought i would patch a simple bug. But Im not sure this is needed. I dont even remember what I did to find this bug. Im closing this since it has not been a priority for me (or anyone else) to fix it.
